micro_sam/training/peft_sam.py

```diff
@@ -0,0 +1,155 @@
+import math
+from typing import Tuple, List, Union
+
+import torch
+import torch.nn as nn
+
+from segment_anything.modeling import Sam
+
+
+class PEFTBase(nn.Module):
+    """PEFTBase is an interface to implement specific PEFT-based methods.
+    """
+    def __call__(
```

Comment:
It would make more sense to have an `__init__` here, that defines a common signature for all PEFT adapter child classes.